modules/terrain/src/lib/parse-terrain.js

```diff
@@ -61,23 +61,43 @@ function getMeshAttributes(vertices, terrain, tileSize, bounds) {
   };
 }
 
-function getMartiniTileMesh(terrainImage, terrainOptions) {
+/**
+ * Returns generated mesh object from image data
+ *
+ * @param {Array} terrainImage terrain image data
+ * @param {object} terrainOptions terrain options
+ * @returns mesh object
+ */
+function getMesh(terrainImage, terrainOptions) {
   if (terrainImage === null) {
     return null;
   }
   const {meshMaxError, bounds, elevationDecoder} = terrainOptions;
 
-  const data = terrainImage.data;
-  const tileSize = terrainImage.width;
-  const gridSize = tileSize + 1;
+  const {data, width, height} = terrainImage;
+  const terrain = getTerrain(data, width, height, elevationDecoder);
 
-  const terrain = getTerrain(data, tileSize, elevationDecoder);
+  let mesh;
 
-  const martini = new Martini(gridSize);
-  const tile = martini.createTile(terrain);
-  const {vertices, triangles} = tile.getMesh(meshMaxError);
+  switch (terrainOptions.tesselator) {
+    case 'martini':
+      mesh = getMartiniTileMesh(meshMaxError, width, terrain);
+      break;
+    case 'delatin':
+      mesh = getDelatinTileMesh(meshMaxError, width, height, terrain);
+      break;
+    // auto
+    default:
+      if (width === height && width !== 0 && !(width & (width - 1))) {
```

Comment:
```suggestion
      if (width === height && !(height & (width - 1))) {
```
You need to check that height vs width is a power of 2, not just width against itself.